jax/_src/ad_checkpoint.py

```diff
@@ -846,7 +846,7 @@ def _optimization_barrier_lowering_rule(ctx, *args):
   barrier_op = hlo.OptimizationBarrierOp(flat_args)
   return mlir.unflatten_ir_values_like_types(barrier_op.results, barrier_types)
 
-def _optimization_barrier(arg):
+def optimization_barrier(arg):
```

Comment:
Shouldn't you delete the abstract_eval rules and lowering rules too? (line 840, 843, etc)

Reply:
I guess I didn't press save in my git checkout on that change.